### PR TITLE
Random mouse placement

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Mice (the rodent, not the peripheral) now start in random locations.
/:cl:

- Adds the SSsqueak subsystem for initial placement of mice, with some mild changes to /mob/living/simple_animal to allow for environment testing (so mice don't get spawned on the solars)

- [x] Edit the maps to remove all starting mice (apart from the named ones in Perma etc.)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -182,66 +182,69 @@
...
-					atmos_suitable = 0
-
-				if(!atmos_suitable)
-					adjustBruteLoss(unsuitable_atmos_damage)
-
-		else
-			if(atmos_requirements["min_oxy"] || atmos_requirements["min_tox"] || atmos_requirements["min_n2"] || atmos_requirements["min_co2"])
-				adjustBruteLoss(unsuitable_atmos_damage)
+	if(!environment_is_safe(environment))
+		adjustBruteLoss(unsuitable_atmos_damage)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: this should probably be adjustHealth so it doesn't pass through brute damage multipliers

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -182,66 +182,69 @@
...
 
 	handle_temperature_damage()
 
 /mob/living/simple_animal/proc/handle_temperature_damage()
-	if(bodytemperature < minbodytemp)
-		adjustBruteLoss(2)
-	else if(bodytemperature > maxbodytemp)
-		adjustBruteLoss(3)
+	if((bodytemperature < minbodytemp) || (bodytemperature > maxbodytemp))
+		adjustBruteLoss(unsuitable_atmos_damage)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: same here

**Comment by ChangelingRain** [comment]
what's the mouse average for all stations combined
please set the amount of mice to the mouse average

**Comment by coiax** [comment]
Some stations have none, some (Dreamstation) have an absolute fucktonne. So I picked Box, which has 9, which seems to be the rough middle? We'll see how it works in practice.

**Comment by ExcessiveUseOfCobblestone** [comment]
is there no better way to do this than to make it a subsystem? I would think since it should only appear once per round [at the start], then it shouldn't need to be a subsystem unless it also spawns mice occasionally when the number is below the threshold.

**Comment by coiax** [comment]
It's an init-only subsystem, such things exist already in the code, like SSjobs

**Comment by CrAzYPiLoT-SS13** [comment]
>SSsqueak
Wew, mate, you really went deep on this one.

**Comment by Cyberboss** [comment]
Don't mean to be that guy, but shouldn't this just be a part of the mapping subsystem?

**Comment by Iamgoofball** [comment]
procedurally generate the mice, the names of the mice, their backstories, and base their placement off of that

**Comment by coiax** [comment]
We don't have a mapping subsystem, map loading is done at startup by BYOND itself.

And SSsqueak was suggested by @RemieRichards, I was going to call it SSmurinae.

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 3]
```diff
@@ -182,66 +182,69 @@
...
-/mob/living/simple_animal/handle_environment(datum/gas_mixture/environment)
-	var/atmos_suitable = 1
+/mob/living/simple_animal/proc/environment_is_safe(datum/gas_mixture/environment, check_temp = FALSE)
+	. = TRUE
 
 	if(pulledby && pulledby.grab_state >= GRAB_KILL && atmos_requirements["min_oxy"])
-		atmos_suitable = 0 //getting choked
+		. = FALSE //getting choked
+
+	if(isturf(src.loc) && isopenturf(src.loc))
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: use 

    var/turf/T  = get_turf(src)
    if(T)

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -182,66 +182,69 @@
...
+		if(ST.air)
+			var/ST_gases = ST.air.gases
+			ST.air.assert_gases(arglist(hardcoded_gases))
+
+			var/tox = ST_gases["plasma"][MOLES]
+			var/oxy = ST_gases["o2"][MOLES]
+			var/n2  = ST_gases["n2"][MOLES]
+			var/co2 = ST_gases["co2"][MOLES]
+
+			ST.air.garbage_collect()
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: What is the purpose of this?

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 3 above)
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: more like `var/turf/open/T = get_turf(src)``

File: code/controllers/subsystem/squeak.dm [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		else
+			M.forceMove(proposed_turf)
+		if(M.environment_is_safe())
+			num_mice -= 1
+			M = null
+
+/datum/subsystem/squeak/proc/find_exposed_wires()
+	exposed_wires.Cut()
+
+	var/all_turfs = block(locate(1,1,1), locate(world.maxx, world.maxy, 1))
```
Comment (optimumtact) [inline code/controllers/subsystem/squeak.dm]: I don't like this at all, is there no other way for this list to be built?

Comment (optimumtact) [inline code/controllers/subsystem/squeak.dm]: Or just spawn the mice into maint areas

**Comment by Cyberboss** [comment]
@coiax https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/mapping.dm

**Comment by optimumtact** [comment]
There are lots of map changes on mini, did you do any remapping?

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 4 above)
Comment (coiax) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: I don't know, it was there before.

File: code/controllers/subsystem/squeak.dm (hunk 5 above)
Comment (coiax) [inline code/controllers/subsystem/squeak.dm]: It runs once, and there are locations I want to include that aren't covered in /area/maintenance. By calculating fresh, we can also reuse the proc for midround spawns.

**Comment by coiax** [comment]
No remapping, seems like the mapmerger was just in a mood.

File: code/controllers/subsystem/squeak.dm (hunk 5 above)
Comment (optimumtact) [inline code/controllers/subsystem/squeak.dm]: I suppose it's fine

Comment (RemieRichards) [inline code/controllers/subsystem/squeak.dm]: it's not fine, you didn't type it as a list! reeeeeee.

Comment (coiax) [inline code/controllers/subsystem/squeak.dm]: fixed

**Pull request merged**